src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -170,8 +161,146 @@ public Task<RestoreResult> ExecuteAsync()
+            {
+                _logger.LogMinimal(string.Format(
+                    CultureInfo.CurrentCulture,
+                    Strings.Log_RestoringToolPackages,
+                    tool.LibraryRange.Name,
+                    _request.Project.FilePath));
+
+                // Build a package spec in memory to execute the tool restore as if it were
+                // its own project. For now, we always restore for a null runtime and a single
+                // constant framework.
+                var toolPackageSpec = new PackageSpec(new JObject())
+                {
+                    Name = string.Empty,
+                    Dependencies = new List<LibraryDependency>(),
+                    TargetFrameworks =
+                    {
+                         new TargetFrameworkInformation
+                        {
+                            FrameworkName = FrameworkConstants.CommonFrameworks.NetStandardApp15,
+                            Dependencies = new List<LibraryDependency>
+                            {
+                                new LibraryDependency
+                                {
+                                    LibraryRange = tool.LibraryRange
+                                }
+                            }
+                        }
+                    }
+                };
+
+                // Execute the restore.
+                var runtimeIds = new HashSet<string>();
+                var projectFrameworkRuntimePairs = CreateFrameworkRuntimePairs(toolPackageSpec, runtimeIds);
+                var allInstalledPackages = new HashSet<LibraryIdentity>();
+                var contextForTool = CreateRemoteWalkContext(_request);
+                var walker = new RemoteDependencyWalker(contextForTool);
+                var result = await TryRestore(tool.LibraryRange,
+                                              projectFrameworkRuntimePairs,
+                                              allInstalledPackages,
+                                              localRepository,
+                                              walker,
+                                              contextForTool,
+                                              writeToLockFile: true,
+                                              token: token);
+
+                var graphs = result.Item2;
+                if (!result.Item1)
+                {
+                    _success = false;
+                }
+
+                // Create the lock file (in memory).
+                var lockFile = CreateLockFile(
+                    previousLockFile: null,
+                    project: toolPackageSpec,
+                    targetGraphs: graphs,
+                    repository: localRepository,
+                    context: contextForTool);
+
+                // Build the path based off of the resolved tool. For now, we assume there is only
+                // ever one target.
+                var target = lockFile.Targets.FirstOrDefault();
```

Comment:
No, never should be single since we always specify the single TFM (`netstandardapp1.5`).